udp_driver/src/udp_component.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 Apex.AI, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "udp_component/udp_component.hpp"
+
+#include <thread>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+
+
+namespace udp_component
+{
+
+UdpComponent::UdpComponent(const rclcpp::NodeOptions & options)
+: UdpDriverT("udp_component", options)
+{
+  RCLCPP_INFO(this->get_logger(), "Initializing udp_component");
+
+  // set output using already declared ROS parameters
+  this->get_parameter("ip", address_);
+  this->get_parameter("port", port_);
+  RCLCPP_INFO(this->get_logger(), "ip: %s", address_.c_str());
+  RCLCPP_INFO(this->get_logger(), "port: %i", port_);
```

Comment:
If these are going to be used outside of constructing the socket, it probaly makes sense to store them as private members of UdpDriverNode when they are retreived as parameters instead of having to `get_parameter` them here.

Reply:
sorry i didnt see that UdpDriverNode already had these as member variables. ill remove the ones I added to UdpComponent.

Reply:
i went ahead and added these as member variables for the UdpDriverNode, along with get functions since they are protected.